pandas/core/internals/blocks.py

```diff
@@ -3210,7 +3210,7 @@ def _putmask_smart(v, m, n):
                 nv = v.copy()
                 nv[m] = nn_at
                 return nv
-    except (ValueError, IndexError, TypeError):
+    except (ValueError, IndexError, TypeError, OverflowError):
```

Comment:
are both of these locations hit by your tests?

Reply:
Yes they are